src/preact/base-element.js

```diff
@@ -209,7 +209,20 @@ export class PreactBaseElement extends AMP.BaseElement {
   isLayoutSupported(layout) {
     const Ctor = this.constructor;
     if (Ctor['layoutSizeDefined']) {
-      return isLayoutSizeDefined(layout);
+      return (
+        isLayoutSizeDefined(layout) ||
+        // This allows a developer to specify the component's size using the
+        // user stylesheet without the help of AMP's static layout rules.
+        // Bento components use `ContainWrapper` with `contain:strict`, thus
+        // if a user stylesheet doesn't provide for the appropriate size, the
+        // element's size will be 0. The user stylesheet CSS can use
+        // fixed `width`/`height`, `aspect-ratio`, `flex`, `grid`, or any
+        // other CSS layouts coupled with `@media` queries and other CSS tools.
+        // Besides normal benefits of using plain CSS, an important feature of
+        // using this layout is that AMP does not add "sizer" elements thus
+        // keeping the user DOM clean.
+        layout == Layout.CONTAINER
```

Comment:
Where is `contain:strict` specified in Bento mode? I think we only have it on AMP mode currently (via the `ampshared.css` class). Meaning Bento mode would behave badly unless the pub was setting it themselves.

Reply:
No, it's specified on the Preact component itself via use of `ContainWrapper`. Basically we have this structure:

```
<amp-comp>
  <comp style="contain:strict">
  </comp>
</amp-comp>
```

The `comp` has 0-size and as the result `amp-comp` also has 0-size. It might, in reality, actually have `width:auto; height: 0`, but for all intents and purposes it's the same as 0-size.